### PR TITLE
feat: implement stencils resource

### PR DESCRIPTION
**Pull request by williansouzagonc**
This PR implements a new resource in our tf provider:

- Stencil

### CODE REVIEW

File: examples/resources/eva_stencil.tf [hunk 1]
```diff
@@ -0,0 +1,31 @@
+resource "eva_stencil" "test" {
+    name            = "my stencil"
+    organization_id = 1
+    language_id     = 1
+    country_id      = 1
+    header          = "<header></header>"
+    template        = "<html></html>"
+    footer          = "<footer></footer>"
+    helpers         = "function someJavascriptFunction() {}; someJavascriptFunction();"
```
Comment (mad-it) [inline examples/resources/eva_stencil.tf]: Can you maybe create an example with multiline? So using 
```tf
<<EOS
<some html>
EOS
```

File: internal/provider/resource_eva_stencil.go [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+				Optional:            true,
+				Type:                types.StringType,
+			},
+			"helpers": {
+				MarkdownDescription: "Helper script for the stencil template",
+				Optional:            true,
+				Type:                types.StringType,
+			},
+			"type": {
+				MarkdownDescription: "Type of the stencil",
```
Comment (mad-it) [inline internal/provider/resource_eva_stencil.go]: Can you add what the type options are?
An example: https://github.com/mad-it/terraform-provider-eva/blob/main/internal/provider/resource_eva_organization_unit.go#L61

File: internal/provider/resource_eva_stencil.go [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+	var data stencilTypeData
+
+	diags := req.Config.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// TODO: do we need an ID or some unique identifier for this resource?
```
Comment (mad-it) [inline internal/provider/resource_eva_stencil.go]:
```suggestion
```

**Review by mad-it — APPROVED** [review]
Looks good, just a few document stuff but the rest looks good.

File: internal/provider/resource_eva_stencil.go (hunk 2 above)
Comment (williansouzagonc) [reply internal/provider/resource_eva_stencil.go]: yeah, nice idea, I've also added a better description for other fields

**Pull request merged**